app/controllers/users.js

```diff
@@ -98,5 +98,17 @@ module.exports = {
         }
       })
       .catch(logDBError(res));
+  },
+  albumList(req, res, next) {
+    if (!req.user.admin && req.user.id !== parseInt(req.params.id)) {
+      return res.status(401).json({ message: 'User is not an admin.' });
+    }
+    Purchase.findAll({ where: { userId: req.params.id } }).then(purchases => {
+      res.status(200).json({
+        albums: purchases.map(p => {
+          return { id: p.albumId };
```

Comment:
This idea of only sending some info is great and it could be moved to a serializer. Anyways in this case since its small logic it can stay here without any problem.